src/posix/platform/coprocessor_type.h

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright (c) 2024, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OT_PLATFORM_COPROCESSOR_MODE_H_
+#define OT_PLATFORM_COPROCESSOR_MODE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Represents the mode of the co-processor.
+ * A co-processor could be either a RCP or NCP.
+ */
+enum CoprocessorType
+{
+    OT_COPROCESSOR_UNKNOWN = 0,
+    OT_COPROCESSOR_RCP     = 1,
+    OT_COPROCESSOR_NCP     = 2,
+};
```

Comment:
This seems to be defined as `C` type, so let's add `typedef` for it
```suggestion
typedef enum CoprocessorType
{
    OT_COPROCESSOR_UNKNOWN = 0,
    OT_COPROCESSOR_RCP     = 1,
    OT_COPROCESSOR_NCP     = 2,
} CoprocessorType;
```